ns1/resource_notifylist.go

```diff
@@ -66,39 +66,40 @@ func notifyListToResourceData(d *schema.ResourceData, nl *monitor.NotifyList) er
 func resourceDataToNotifyList(nl *monitor.NotifyList, d *schema.ResourceData) error {
 	nl.ID = d.Id()
 
-	if rawNotifications := d.Get("notifications").([]interface{}); len(rawNotifications) > 0 {
-		ns := make([]*monitor.Notification, len(rawNotifications))
-		for i, notificationRaw := range rawNotifications {
+	index := 0
+	if rawNotifications := d.Get("notifications").(*schema.Set); rawNotifications.Len() > 0 {
+		ns := make([]*monitor.Notification, rawNotifications.Len())
```

Comment:
I think if you set this to
```
ns := make([]*monitor.Notification, 0, rawNotifications.Len())
```
then you don't need to define the index separately from the for loop. Then change the additions to the slice to appends, e.g. 
```
ns[index] = monitor.NewEmailNotification(email.(string))
// change to:
ns = append(ns, monitor.NewEmailNotification(email.(string)))
```

Reply:
Thanks! Going to test it next week